x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,529 @@
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
+Create a new PKCS#12 file that contains a certificate and key and replace
+the existing PKCS#12 file (such as `elastic-certificates.p12`) with the new
+certificate. 
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+security certificates that are currently in use. You'll use the same names
+for your new certificate.
++
+In this example, the keystore and truststore are pointing to different files.
+Your configuration might use the same file that contains the certificate and CA.
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.keystore.password: password
+xpack.security.transport.ssl.truststore.path: config/elastic-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+xpack.security.transport.ssl.truststore.password: password
+----
+
+. On any node in your cluster, generate a certificate and private key for your
+node. You must include the CA that was used to sign the certificate currently
+in use.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `--ca <ca_file>`:: Name of the CA file used to sign your certificates. The
+   default file name from the `elasticsearch-certutil` tool is
+   `elastic-stack-ca.p12`.
+====
++
+    a. Enter the password for your CA when prompted.
+
+    b. Enter the name for your certificate, which must match the name of the
+    certificate currently in use.
+
+    c. (Optional) Enter a password for your certificate. 
+
+. If you entered a password when creating the node certificate, run the following commands to store the password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.truststore.secure_password
+----
+--
+
+. [[cert-password-updates]] If your certificate is password protected, the password is stored in the {es} keystore, _and_ the password needs to change, then you must perform a <<restart-cluster-rolling,rolling restart>> on your cluster. Complete the steps until the step that indicates *Perform any needed changes*, and then continue to the next step to replace your existing certificates.
+
+. Replace your existing certificate with the new certificate, ensuring that the
+file names match.
+
+. Repeat the previous step on each node in your cluster to update the existing certificate, completing a rolling restart beforehand if your certificate matches the password stipulations outlined in <<cert-password-updates,this step>>.
```

Comment:
certificiate => "keystore" or "key and certificate"

Reply:
They don't need to perform a rolling restart after the changes in every node. They need to restart the node after the changes in every node, which - after doing it for all nodes - will constitute a rolling restart of the cluster

Reply:
I think that these steps provide greater clarity:

> . On every node in your cluster, replace your existing keystore with the new keystore, ensuring that the file names match. For example, `elastic-stack-ca.p12`.
> 
> . After replacing the keystore on every node in your cluster, restart each node one at a time.
> 
> . If your keystore password changed and you started a rolling restart in  <<cert-password-updates,this step>>, complete the remaining steps to finish the rolling restart.

Reply:
> They don't need to perform a rolling restart after the changes in every node. They need to restart the node after the changes in every node, which - after doing it for all nodes - will constitute a rolling restart of the cluster

But if they're using the same file name, they **do not** need to restart the node, correct? Or do they need to restart the node regardless of whether they're using the same file name of the existing keystore file?